Services/QueryBuilderService.cs

```
@@ -18,27 +18,18 @@ public Dictionary<string, object[]> RunQuery(Query query)
                 .FirstOrDefault(p => p.PropertyType.IsGenericType && p.Name.ToLower(CultureInfo.InvariantCulture) == t && p.PropertyType.GetGenericTypeDefinition() == typeof(DbSet<>));
             if (propInfo != null)
             {
-                var methods = typeof(QueryExecutor).GetMethods(BindingFlags.Static | BindingFlags.Public);
-                var method = methods?.FirstOrDefault(m => m.CustomAttributes.Count() == 2);
-
                 var dbSet = propInfo.GetValue(db);
                 var dbGenericType = dbSet?.GetType()?.GenericTypeArguments.FirstOrDefault();
-                if (dbGenericType != null && dbSet != null)
+                var dtoGenericType = typeof(TResults).GetProperty(propInfo.Name)?.PropertyType.GetElementType();
+
+                if (dbSet != null && dbGenericType != null && dtoGenericType != null)
```

Throw instead of silently ignoring errors.

```cs
var resultProperty = typeof(TResults).GetProperty(t, BindingFlags.Instance | BindingFlags.Public | BindingFlags.IgnoreCase)
    ?? throw new ValidationException($"Unknown entity {t}");

var dtoGenericType = resultProperty.PropertyType.GetElementType();
```

```cs
var dbSet = propInfo.GetValue(db)
    ?? throw new ValidationException($"propInfo.Name dbSet is null");

var dbGenericType = dbSet.GetType().GenericTypeArguments.FirstOrDefault()
    ?? throw new ValidationException($"Missing DbSet generic type");
```

---

Fixed.